packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -746,6 +766,7 @@ export class LocalFluidDataStoreContext extends LocalFluidDataStoreContextBase {
          * @deprecated 0.16 Issue #1635, #3631
          */
         createProps?: any,
+        isRootDataStore: boolean = false,
```

Comment:
I'd make it non-optional arg - force caller to make a choice, not fall victim to ignorance